src/main/kotlin/net/lumalyte/lg/infrastructure/services/PartyServiceBukkit.kt

```
@@ -412,6 +416,63 @@ class PartyServiceBukkit(
     }
     
+    override fun removeGuildFromPartyAsSystem(partyId: UUID, guildId: UUID): Party? {
+        return try {
+            val party = fetchActivePartyWithGuild(partyId, guildId) ?: return null
+            val remainingGuilds = party.guildIds - guildId
+            if (remainingGuilds.size < MINIMUM_PARTY_GUILDS) {
+                dissolvePartyAfterRemoval(party, guildId)
+            } else {
+                updatePartyAfterRemoval(party, remainingGuilds, guildId)
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)